server/log_rpc_server.go

```diff
@@ -78,12 +78,6 @@ func (t *TrillianLogRPCServer) QueueLeaves(ctx context.Context, req *trillian.Qu
 	}
 	leaves := depointerify(req.Leaves)
 
-	// TODO(al): Hasher must be selected based on log config.
-	th, _ := merkle.Factory(merkle.RFC6962SHA256Type)
```

Comment:
I think we should make the personality supply all the data in `LogLeaf`. 
This allows personalities to define their own leaf hashing algorithms without needing to change server code. What do you think?